### PR TITLE
Ensure OIDs are unique to each DB

### LINKED ISSUE #559: Duplicate tables are present with the same oid in mathesar_table
## Description
Duplicate tables are reflected in `mathesar_table` with the same oid, when trying to import a file. This is not regular. Happens intermittently.

Screenshot 2021-08-15 at 8 09 57 PM

The actual database only contains one table with that oid, as expected.

To note, whenever this issue occurs, one of the entries has a different value for `import_verified`.

This leads to the following error, when reflection happens the next time.

```
Request Method: GET
Request URL: http://localhost:8000/mathesar_tables/1/

Django Version: 3.1.7
Python Version: 3.9.6
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'rest_framework',
 'django_filters',
 'django_property_filter',
 'mathesar']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/code/mathesar/views/frontend.py", line 83, in schema_home
    'common_data': get_common_data(request, database, schema)
  File "/code/mathesar/views/frontend.py", line 29, in get_common_data
    'schemas': get_schema_list(request, database),
  File "/code/mathesar/views/frontend.py", line 9, in get_schema_list
    Schema.objects.filter(database=database),
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/code/mathesar/models.py", line 25, in get_queryset
    reflection.reflect_db_objects()
  File "/code/mathesar/reflection.py", line 99, in reflect_db_objects
    reflect_tables_from_schema(schema)
  File "/code/mathesar/reflection.py", line 60, in reflect_tables_from_schema
    models.Table.current_objects.get_or_create(oid=oid, schema=schema)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 573, in get_or_create
    return self.get(**kwargs), False
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 433, in get
    raise self.model.MultipleObjectsReturned(

Exception Type: MultipleObjectsReturned at /mathesar_tables/1/
Exception Value: get() returned more than one Table -- it returned 2!
```

## Expected behavior
Duplicate table entries should not be present.

## To Reproduce
This issue happens intermittently, and I've not been able to reproduce it. It has occurred twice on my local instance.

### PR DESCRIPTION
**Pull request by eito-fis**
Fixes #559

Adds constraints and validation such that OID is unique on a per-db basis.

**Technical details**
- Adds a composite unique constraint to the `Schema` model that ensures OID and database are unique together. 
- Adds unique validation to the `Table` model that ensures OID and database are unique together. This requires a `validate_unique` method that we ensure is called in the `save` method. We can't use a `UniqueConstraint` here as we do not have access to the database object directly, only as a field of the related `Schema`.

Additionally, I updated the table patch endpoint's `Table.save()` call to use the `update_fields` argument. This should ensure that the patch endpoint only ever updates, which hopefully solves the original issue of duplicate tables.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/models.py [hunk 1]
```diff
@@ -127,6 +133,19 @@ class Table(DatabaseObject):
                                related_name='tables')
     import_verified = models.BooleanField(blank=True, null=True)
 
+    def validate_unique(self, exclude=None):
```
Comment (kgodey) [inline mathesar/models.py:136]: I would call this `_validate_unique` to indicate that it's an internal method.

**Review by kgodey — APPROVED** [review]
@eito-fis Looks great, thanks! I added one small review comment that you can fix before merge.

File: mathesar/models.py (hunk 1 above)
Comment (eito-fis) [reply mathesar/models.py:136]: [`validate_unique`](https://docs.djangoproject.com/en/3.2/ref/models/instances/#django.db.models.Model.validate_unique) is actually a defined django method. I think it is most often used in the context of a form, where you call `full_clean`, but still seems somewhat idiomatic here. Happy to change if you prefer the internal method.

Comment (kgodey) [reply mathesar/models.py:136]: Oh, it is? Let's use it then.

**Pull request merged**